bcipy/tasks/rsvp/stopping_criteria.py

```diff
@@ -0,0 +1,231 @@
+        cur_dist = epoch['list_distribution'][-1]
+        return np.argmax(cur_dist) == np.argmax(
+            prev_dist) and np.max(cur_dist) < np.max(prev_dist)
+
+
+class MaxIterationsCriteria(DecisionCriteria):
+    """Returns true if the max iterations have been reached."""
+
+    def __init__(self, max_num_seq):
+        """ Args:
+                max_num_seq(int): maximum number of sequences allowed before
+                    mandatory termination """
+        self.max_num_seq = max_num_seq
+
+    def decide(self, epoch):
+        # Note: len(epoch['list_sti']) != len(epoch['list_distribution'])
+        # see MinIterationsCriteria comment
+        current_seq = len(epoch['list_sti'])
+        if current_seq >= self.max_num_seq:
+            log.debug(
+                "Committing to decision: max iterations have been reached.")
+            return True
+        return False
+
+
+class ProbThresholdCriteria(DecisionCriteria):
+    """Returns true if the commit threshold has been met."""
+
+    def __init__(self, threshold):
+        """ Args:
+                threshold(float \in [0,1]): A threshold on most likely
+                    candidate posterior. If a candidate exceeds a posterior
+                    the system terminates.
+                 """
+        assert 1 >= threshold >= 0, "stopping threshold should be in [0,1]"
+        self.tau = threshold
+
+    def decide(self, epoch):
+        current_distribution = epoch['list_distribution'][-1]
+        if np.max(current_distribution) > self.tau:
+            log.debug("Committing to decision: posterior exceeded threshold.")
+            return True
+        return False
+
+
+class MarginCriteria(DecisionCriteria):
+    """ Stopping criteria based on the difference of two most likely candidates.
+        This condition
+    """
+
+    def __init__(self, margin):
+        """ Args:
+                margin(float \in [0,1]): Minimum distance required between
+                    two most likely competing candidates to trigger termination.
+                    """
+        assert 1 >= margin >= 0, "difference margin should be in [0,1]"
+        self.margin = margin
+
+    def decide(self, epoch):
+        # Get the current posterior probability values
+        p = copy(epoch['list_distribution'][-1])
+        # This criteria compares most likely candidates (best competitors)
+        candidates = [p[idx] for idx in list(np.argsort(p)[-2:])]
+        stopping_rule = np.abs(candidates[0] - candidates[1])
+        d = stopping_rule > self.margin
+        if d:
+            log.debug("Committing to decision: margin is high enough.")
+
+        return d
+
+
+class MomentumCommitCriteria(DecisionCriteria):
+    """ Stopping criteria based on Renyi entropy on the simplex
+        Renyi entropy for infinity and 1 norms are special forms.
+        Attr:
+            alpha(float): a non-negative order for the entropy
+            tau(float): decision threshold
+            """
```

Comment:
The documentation does not match the parameters here.

Reply:
This documentation was also fixed.